pandas/tests/indexes/test_multi.py

```diff
@@ -3307,3 +3307,40 @@ def test_duplicate_multiindex_labels(self):
         with pytest.raises(ValueError):
             ind.set_levels([['A', 'B', 'A', 'A', 'B'], [2, 1, 3, -2, 5]],
                            inplace=True)
+
+    def test_use_enum_in_multiindex(self):
+        # GH 21298
+        # Allow use of Enums as one of the factors in a MultiIndex.
+
+        # See https://github.com/pandas-dev/pandas/blob/
+        # 0c65c57a279e755ab7093db925d1e580f9878dae/pandas/util/testing.py#L793-L799.
+        # We cannot simply use tm.assert_index_equal, as the "expected" index
+        # cannot actually be built yet.
+        from pandas.core.algorithms import take_1d
+
+        def _get_ilevel_values(index, level):
+            # accept level number only
+            unique = index.levels[level]
+            labels = index.labels[level]
+            filled = take_1d(
+                unique.values,
+                labels, fill_value=unique._na_value
+            )
+            values = unique._shallow_copy(filled, name=index.names[level])
+            return values
+
+        from enum import Enum
+        MyEnum = Enum("MyEnum", "A B")
```

Comment:
you would have to add ``enum34`` (the 2.7 backport) in our 2.7 builds

```
(pandas) bash-3.2$ grep python=2.7 ci/*.yaml
appveyor-27.yaml:  - python=2.7.*
circle-27-compat.yaml:  - python=2.7*
travis-27-locale.yaml:  - python=2.7
travis-27.yaml:  - python=2.7*
```

Reply:
I'm sorry, does this mean simply importing `enum34` inline? Thank you.

Reply:
ok, I've hopefully done this correctly.